lib/async-emitter.js

```diff
@@ -53,8 +53,22 @@ class AsyncEmitter {
   }
 
   clear(name) {
-    if (name) this.events.delete(name);
-    else this.events.clear();
+    if (name) {
+      const event = this.events.get(name);
+      if (event) {
+        for (const wrapper of event.values()) {
+          for (const [handler, fn] of this.wrappers.entries()) {
+            if (wrapper === fn) {
+              this.wrappers.delete(handler);
```

Comment:
@lundibundi here is nested loops

Reply:
I'll remove it in next commit

Reply:
```js
if (!name) {
  this.wrappers.clear();
  this.events.clear();
  return;
}
const events = this.events.get(name);
if (events) {
  this.wrappers.values().forEach(wrapper => {
    if (events.has(wrapper)) this.wrappers.delete(wrapper);
  })
}
this.events.delete(name);
```
?

Reply:
Good solution, except there is no `forEach` in Iterator and we don't need to `this.events.delete(name);` if no event found. See next changes: